trunk/SUAVE/Analyses/Aerodynamics/AVL_Inviscid.py

```diff
@@ -0,0 +1,338 @@
+
+# SUAVE imports
+import SUAVE
+from SUAVE.Core import Units, Data
+from SUAVE.Core import redirect
+
+from SUAVE.Analyses.Mission.Segments.Conditions.Aerodynamics import Aerodynamics
+from SUAVE.Analyses.Mission.Segments.Conditions.Conditions   import Conditions
+
+from SUAVE.Methods.Aerodynamics.AVL.write_geometry   import write_geometry
+from SUAVE.Methods.Aerodynamics.AVL.write_run_cases  import write_run_cases
+from SUAVE.Methods.Aerodynamics.AVL.write_input_deck import write_input_deck
+from SUAVE.Methods.Aerodynamics.AVL.run_analysis     import run_analysis
+from SUAVE.Methods.Aerodynamics.AVL.translate_data   import translate_conditions_to_cases, translate_results_to_conditions
+from SUAVE.Methods.Aerodynamics.AVL.purge_files      import purge_files
+from SUAVE.Methods.Aerodynamics.AVL.Data.Results     import Results
+from SUAVE.Methods.Aerodynamics.AVL.Data.Settings    import Settings
+from SUAVE.Methods.Aerodynamics.AVL.Data.Cases       import Run_Case
+
+# Package imports
+import time
+import pylab as plt
+import os
+import numpy as np
+import sys
+import sklearn
+from sklearn import gaussian_process
+from sklearn import neighbors
+from sklearn import svm
+from shutil import rmtree
+from warnings import warn
+
+## ----------------------------------------------------------------------
+##  Class
+## ----------------------------------------------------------------------
+
+class AVL_Inviscid(Aerodynamics):
+    """ SUAVE.Analyses.Aerodynamics.AVL
+        aerodynamic model that performs a vortex lattice analysis using AVL
+        (Athena Vortex Lattice, by Mark Drela of MIT).
+
+        this class is callable, see self.__call__
+
+    """
+
+    def __defaults__(self):
+        self.tag        = 'avl'
+        self.keep_files = True
+
+        self.settings = Settings()
+
+        self.current_status = Data()
+        self.current_status.batch_index = 0
+        self.current_status.batch_file  = None
+        self.current_status.deck_file   = None
+        self.current_status.cases       = None
+        
+        self.geometry = None
+        
+        self.settings.filenames.log_filename = sys.stdout
+        self.settings.filenames.err_filename = sys.stderr
+        
+        # Conditions table, used for surrogate model training
+        self.training = Data()   
+        
+        #HALE UAV Cruise
+        self.training.angle_of_attack  = np.array([0 , 2, 4, 6 ,8, 10]) * Units.deg
+        self.training.Mach             = np.array([0.075, 0.1, 0.125 ,0.15,0.175, 0.2])       
+    
+        # Standard subsonic/transolic aircarft
+        #self.training.angle_of_attack  = np.array([-2.,3.,8.])
+        #self.training.Mach             = np.array([0.3,0.7,0.85]) 
+        
+        self.training.lift_coefficient = None
+        self.training.drag_coefficient = None
+        self.training_file             = None
+        
+        # Surrogate model
+        self.surrogates = Data()
+        self.surrogates.moment_coefficient = None
```

Comment:
I don't think the moment calculations belong in this module - there are several in this file.